### ISSUE TITLE
Komodo Edit loses focus

### ISSUE THREAD

**Issue by jsnjack** [Komodo/KomodoEdit#1040]
I'm using Komodo Edit 9.3.2 build 16460 (platform linux-x86_64, running on Linux 4.3.4-300.fc23.x86_64 version #1 SMP Mon Jan 25 13:39:23 UTC 2016)

I noticed that sometimes komodo just loses focus. Application becomes totally unresponsive. No menus or key-bindings work, even close button. Unfortunately, I can't find a way to reproduce it because it always happens in the middle of the working process with application

**Comment by mitchell-as** [comment]
Please let us know when you manage to reproduce. That will help us in figuring out and fixing the issue. Also, whenever you force-kill Komodo, and before you re-open it, check the contents of your error log (~/.komodoide/9.3/pystderr.log) and send it to us. That may also help us identify the problem.

**Comment by jsnjack** [comment]
@mitchell-as I think I found a way to reproduce it:
1. Open project in komodo. Project should contain at least 2 folders
2. Open file from folder 1
3. In the tree view drag folder 1 to the folder 2
4. The interface will be blocked

Also log file:
[pystderr.txt](https://github.com/Komodo/KomodoEdit/files/115627/pystderr.txt)

**Comment by mitchell-as** [comment]
Hi, I'm not able to reproduce this on 64-bit Linux.

1. I created a new project.
2. In the new project, I created two folders "bar" and "foo".
3. In "bar" I created an empty "file1.txt".
4. In "foo" I created an empty "file2.txt".
5. I close both opened files and then re-opened "bar/file1.txt".
6. I dragged "bar" into "foo" in the tree view.
7. I got a popup error, but not a blocked interface. When I put the focus back in the editor view, Komodo notifies me the file no longer exists on disk. No matter which option I pick I still get a responsive Komodo.

Thanks for submitting your error log. It contains exactly the popup error message I saw, but no indication of a hang.

**Comment by jsnjack** [comment]
Hmm.. Strange. For me it is 100% reproducible. Maybe it is something related to the linux distro? I'm using fedora 23 64bit with latest updates 

I also added some screenshots:
1. Project with opened file. I'm going to drag schemas folder to the locale folder in the tree view
![screenshot from 2016-02-03 17-53-22](https://cloud.githubusercontent.com/assets/5493713/12789698/30245a96-ca9f-11e5-90a1-bc6d88e63cf3.png)
2. When I do that I get an error
![screenshot from 2016-02-03 17-53-36](https://cloud.githubusercontent.com/assets/5493713/12789762/7b259a50-ca9f-11e5-8922-402d2a5816de.png)
3. I Click OK
4. When I click on the opened file, Komodo notifies me that file does not exist anymore
![screenshot from 2016-02-03 17-53-46](https://cloud.githubusercontent.com/assets/5493713/12789792/9856fb3c-ca9f-11e5-820f-8945e664006c.png)
5. If I press Cancel button, everything is ok. But if I press OK button, it closes the file and the window is locked

**Comment by Naatan** [comment]
I'm confused, the issue you are explaining there does not seem to talk about loosing focus at all, instead you talk about a locked window? Is this the same issue or are you talking about a different issue?

**Comment by jsnjack** [comment]
Sorry for the confusion. I don't really know how to name it. When I press ok button on the image 4, the main Komodo window becomes unresponsive. I try to click on menu, click close button, do something in the editor or tree view and it does not respond to any my actions. However, it does not give my cup 100% load

**Comment by Naatan** [comment]
Alright, so to be clear - that is the issue you were talking about from the beginning?

Does it always happen on the same file or files? (ie. files residing in the jack@.. folder).

**Comment by jsnjack** [comment]
Yes, that is the issue

No folders and files are different

**Comment by Naatan** [comment]
So regardless of the folder location, when you drag it on places Komodo hangs?

Does it do the same thing when using the `Cog icon > Open Directory` option to open the same folder?

**Comment by jsnjack** [comment]
Yes, it does not depend on the folder location. After I kill komodo and start it again I'm able to open folder without any problems

**Comment by Naatan** [comment]
> Does it do the same thing when using the Cog icon > Open Directory option to open the same folder?

**Comment by jsnjack** [comment]
Komodo UI is blocked. I can't press any buttons.

I uploaded some video. After popup from 4 I'm trying to click on UI objects and get to response

https://drive.google.com/file/d/0Bzf1ot7c1tUscF9GZUJzMC1hbU0/view?usp=sharing

**Comment by Naatan** [comment]
Ahh thanks for that! I was under the false impression that you were dragging a folder onto Komodo from an external source.

I'm able to reproduce the error but not Komodo hanging. Could you try Help > Troubleshooting > Restart in Safe-mode.

**Comment by jsnjack** [comment]
Yes, I should have started with uploading the video. (Had no idea the feature is included with gnome)

I tried Safe-mode, the same issue

**Comment by Naatan** [comment]
I'll look into resolving the error, hopefully that will also resolve the hanging issue but as I cannot reproduce it I'll need you to test it (once we get a nightly up).

**Comment by jsnjack** [comment]
ok, thanks!

**Comment by ckris** [comment]
I have the same problem on windows 10 with version of Komodo IDE 11.
In task manager when focus loses I see crash reporting task for 500ms and after is hide.

[komodo.log](https://github.com/Komodo/KomodoEdit/files/1403213/komodo.log)
